docs/design/provisioning.md

```diff
@@ -0,0 +1,143 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Abstract
+
+This document proposes a model for the configuration and management of dynamically provisioned persistent volumes in Kubernetes.  Familiarity [Persistent Volumes](../user-guide/persistent-volumes/) is assumed.
+
+Administrators use ConfigMap in a namespace they control to configure many provisioners.  Users request volumes via a selector and/or storage class.
+
+## ConfigMap
+
+Provisioners will be configured using `ConfigMap`.  Configs will be held in a namespace controlled by the administrator.
+
+The admin configures kube-controller-manager with a CLI flag specifying the namespace containing the configs.
+
+`--pv-provisioner-namespace=foobar`
+
+
+### Example Configuration
+
+
+```
+	config := &api.ConfigMap {
+		ObjectMeta: api.ObjectMeta{
+			Name: "gold",
+			Namespace: "foobar",
+			Labels: {
+				"storage-class": "gold",
			}
+		},
+		Data: map[string]string {
+			"plugin-name": "kubernetes.io/gce-pd",
+			"service-account-name": "foo",
+			"params": "${OPAQUE-JSON-MAP}",
		}
+	}
+		
+
+	claim := &api.PersistentVolumeClaim{
+		Spec: api.PersistentVolumeClaimSpec{
+				PersistentVolumeSelector:map[string]string{
+					"storage-class":"gold",
+				},
+			},
+		},
+	}
+	
+```
+
+#### Design Consideration
+
+`StorageClass` has been proposed as a field on PersistentVolumeClaim.  An equivalent of that field is currently stewing as an annotation in the current version of dynamic provisioning.
+
+`PersistentVolumeSelector` on PersistentVolumeClaim has been proposed as a means to select specific volumes, akin to a pod's NodeSelector.
+
+Having both fields is confusing.  Consider this on-premise installation of Kube:
+
+> Admin Joe configures "Gold" to dynamically provision Ceph volumes.  Joe also creates "Silver" volumes by manually provisioning a large number of NFS exports on expensive hardware (SSDs, highly available, stringent backup policy).  Lastly, Joe creates "Bronze" NFS exports on older hardware with slower discs.
+
```

Comment:
The requirement for 1 path is noted and a good one to adhere to.

@derekwaynecarr is doing quota and it might be that a single field on PVC is easier for that effort, as opposed to parsing arbitrary labels.

It might be that we need to create a Provisioner plugin that's not part of any volume plugin.  It's behavior is to pull from the pool of available volumes.  Essentially, we would package the existing controller's behavior into a plugin and make manually provisioned storage explicit (use this plugin for this class) as opposed to implicit (no configmap found w/ that label set)